csharp/src/Ice/Communicator.cs

```diff
@@ -360,7 +373,116 @@ private class ObserverUpdater : Instrumentation.IObserverUpdater
 
                 TraceLevels = new TraceLevels(this);
 
-                DefaultsAndOverrides = new DefaultsAndOverrides(this, Logger);
+                DefaultCollocationOptimized = (GetPropertyAsInt("Ice.Default.CollocationOptimized") ?? 1) > 0;
+
+                if (GetProperty("Ice.Default.Encoding") is string encoding)
+                {
+                    DefaultEncoding = Encoding.Parse(encoding);
+                    DefaultEncoding.CheckSupported();
```

Comment:
Same for CheckSupported. It's a configuration error to set DefaultEncoding to a non-supported encoding.

Reply:
Fair enough, fixed!